src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx

```diff
@@ -58,6 +70,64 @@ export const DiscoverTopNav = ({
     [history, resetSavedSearch, savedSearch.id]
   );
 
+  useEffect(() => {
+    return () => {
+      // Make sure to close the editors when unmounting
+      if (closeFieldEditor.current) {
+        closeFieldEditor.current();
+      }
+      if (closeDataViewEditor.current) {
+        closeDataViewEditor.current();
+      }
+    };
+  }, []);
+
+  const editField = useMemo(
+    () =>
+      canEditDataViewField
+        ? async (fieldName?: string, uiAction: 'edit' | 'add' = 'edit') => {
+            if (indexPattern?.id) {
+              const indexPatternInstance = await data.dataViews.get(indexPattern?.id);
```

Comment:
you shouldn't need the optional chaining here `data.dataViews.get(indexPattern?.id);` since you already check `indexPattern?.id`

Reply:
Done